src/TokenizerBase.cpp

```diff
@@ -161,9 +161,10 @@ TokenizerBase::process_number(std::string &val)
 				continue;
 			}
 		}
+        // Underscore is for Java and single quote for C++ digit separators
 		if (!isalnum(c0) && c0 != '.' && c0 != '_' && c0 != '\'')
 			break;
-        if (c0 != '\'')
+        if (c0 != '\'' && c0 != '_' && c0 != '.')
```

Comment:
I think we should also test for '.'

Reply:
I don't think we should test for `.`.   I assume your original idea for adding the test, was to remove decorative elements (as we remove comments), which for numbers are `_` and `'`.